src/models.py

```diff
@@ -102,7 +102,28 @@ def build_model(args, vocab, pretrained_embs, tasks):
     rnn_params = Params({'input_size': d_emb, 'bidirectional': True,
                          'hidden_size': args.d_hid, 'num_layers': args.n_layers_enc})
 
-    if any(isinstance(task, LanguageModelingTask) for task in tasks) or \
+    if args.sent_enc == "onlstm":
+        phrase_layer = ONLSTMStack(
+            [args.d_word] + [args.d_hid] * (args.n_layers_enc - 1) + [args.d_word],
+            chunk_size=args.chunk_size,
+            dropconnect=args.dropconnect,
+            dropouti=args.dropouti,
+            dropout=args.dropout,
+            dropouth=args.dropouth,
+            embedder=embedder,
+            phrase_layer=None,
+            batch_size=args.batch_size
+        )
+        sent_encoder = SentenceEncoder(vocab, embedder, args.n_layers_highway,
+                                       phrase_layer, skip_embs=args.skip_embs,
+                                       dropout=args.dropout,
+                                       sep_embs_for_skip=args.sep_embs_for_skip,
+                                       cove_layer=cove_layer)
+        d_sent = args.d_word
+        log.info("Using ON-LSTM sentence encoder!")
+        if tasks[0].name == "wsj" and args.sent_enc == 'onlstm' and (args.tying == 1):
```

Comment:
Add a comment explaining this.

Reply:
Removed the tying sub-code, as it was making no/little difference in parsing performance.